content/blog/2018/01/2018-01-13-jep-200.adoc

```diff
@@ -0,0 +1,104 @@
+Link it to link:https://issues.jenkins-ci.org/browse/JENKINS-47736[JENKINS-47736] and add the `JEP-200` label.
+If at all possible, include complete steps to reproduce the problem from scratch.
+Jenkins developers will strive to evaluate the reason for the violation and offer a fix in the form of a core and/or plugin update.
+For more details and current status, see
+link:https://wiki.jenkins.io/display/JENKINS/Plugins+affected+by+fix+for+JEP-200[Plugins affected by fix for JEP-200].
+
+Assuming you see no particular reason to think that the class in question has dangerous deserialization semantics, which is rare,
+it is possible to work around the problem in your own installation as a temporary expedient.
+Simply make note of any class name(s) mentioned in such log messages,
+and run Jenkins with this startup option (details will depend on your installation method):
+
+----
+-Dhudson.remoting.ClassFilter=some.pkg.and.ClassName,some.pkg.and.OtherClassName
+----
+
+== For plugin developers
+
+As a plugin developer encountering this kind of error,
+your first task is to ensure that it is reproducible in a functional (`JenkinsRule`) test
+when running
+
+[source,sh]
+----
+mvn test -Djenkins.version=2.102
+----
+
+or the next LTS version of Jenkins reported to reproduce the error (or a newer version).
+The above assumes you are using a recent 2.x parent POM.
+For certain cases you may need to use `plugin-compat-tester` to run tests against Jenkins core versions much newer than your baseline.
+
+You may need to run tests using an agent (e.g., `JenkinsRule.createSlave`) or force saves of plugin settings.
+
+Also make sure your `Jenkinsfile` specifies
+
+[source,groovy]
+----
+buildPlugin(jenkinsVersions: [null, '2.102'])
+----
+
+(again picking whatever version you need to test against)
+so that the test is included during CI builds, even while your minimum core baseline predates JEP-200.
+
+You then have several choices for fixing the problem:
+
+* If the class(es) are defined in the Java Platform or some library bundled in Jenkins core, propose a pull request adding it to `core/src/main/resources/jenkins/security/whitelisted-classes.txt` in `jenkinsci/jenkins`.
+* If the class(es) are defined in a third-party library bundled in your plugin, create a resource file `META-INF/hudson.remoting.ClassFilter` listing them. (link:https://github.com/jenkinsci/workflow-support-plugin/pull/50/files[example])
+** You may also do this for Java or Jenkins core library classes, as a hotfix until your core baseline includes the whitelist entry proposed above.
+* If the class(es) are defined in a JAR you build and then bundle in your plugin’s `*.jpi`, add a `Jenkins-ClassFilter-Whitelisted: true` manifest entry. This whitelists every class in the JAR. (link:https://github.com/jenkinsci/lib-jenkins-maven-embedder/pull/15/files[example])
+* Ideally, simplify your code so that the mentioned class is not deserialized via Jenkins Remoting or XStream to begin with:
```

Comment:
Should probably be the first item and emphasized that this is the preferred approach.